4/lib/model.d.ts

```diff
@@ -0,0 +1,2422 @@
+   * must be true. Not affected by underscored setting.
+   */
+  updatedAt?: string | boolean;
+
+  /**
+   * Defaults to pluralized model name, unless freezeTableName is true, in which case it uses model name
+   * verbatim
+   */
+  tableName?: string;
+
+  /**
+   * Provide getter functions that work like those defined per column. If you provide a getter method with
+   * the
+   * same name as a column, it will be used to access the value of that column. If you provide a name that
+   * does not match a column, this function will act as a virtual getter, that can fetch multiple other
+   * values
+   */
+  getterMethods?: ModelGetterOptions;
+
+  /**
+   * Provide setter functions that work like those defined per column. If you provide a setter method with
+   * the
+   * same name as a column, it will be used to update the value of that column. If you provide a name that
+   * does not match a column, this function will act as a virtual setter, that can act on and set other
+   * values, but will not be persisted
+   */
+  setterMethods?: ModelSetterOptions;
+
+  /**
+   * Provide functions that are added to each instance (DAO). If you override methods provided by sequelize,
+   * you can access the original method using `this.constructor.super_.prototype`, e.g.
+   * `this.constructor.super_.prototype.toJSON.apply(this, arguments)`
+   */
+  instanceMethods?: Object;
+
+  /**
+   * Provide functions that are added to the model (Model). If you override methods provided by sequelize,
+   * you can access the original method using `this.constructor.prototype`, e.g.
+   * `this.constructor.prototype.find.apply(this, arguments)`
+   */
+  classMethods?: Object;
+
+  schema?: string;
+
+  /**
+   * You can also change the database engine, e.g. to MyISAM. InnoDB is the default.
+   */
+  engine?: string;
+
+  charset?: string;
+
+  /**
+   * Finaly you can specify a comment for the table in MySQL and PG
+   */
+  comment?: string;
+
+  collate?: string;
+
+  /**
+   * Set the initial AUTO_INCREMENT value for the table in MySQL.
+   */
+  initialAutoIncrement?: string;
+
+  /**
+   * An object of hook function that are called before and after certain lifecycle events.
+   * The possible hooks are: beforeValidate, afterValidate, beforeBulkCreate, beforeBulkDestroy,
+   * beforeBulkUpdate, beforeCreate, beforeDestroy, beforeUpdate, afterCreate, afterDestroy, afterUpdate,
+   * afterBulkCreate, afterBulkDestory and afterBulkUpdate. See Hooks for more information about hook
+   * functions and their signatures. Each property can either be a function, or an array of functions.
+   */
+  hooks?: HooksOptions;
+
+  /**
+   * An object of model wide validations. Validations have access to all model values via `this`. If the
+   * validator function takes an argument, it is asumed to be async, and is called with a callback that
+   * accepts an optional error.
+   */
+  validate?: ModelValidateOptions;
+
+}
+
+export abstract class Model {
```

Comment:
What if we make `Model` take a type param which is the child class. i.e.

```ts
class User extends Model<User> {
}
```

Would that solve the issue in `findAll`?